File: java/gpt/p4_fc.java
/**
 * p4_fc.java - Safe vehicle index lookup (Java)
 *
 * Prompts the user to enter the index of a vehicle, reads the input safely
 * as a string, attempts to parse it as an integer, validates bounds, and
 * prints the corresponding vehicle from a predefined array. Invalid input
 * or out-of-range indices produce clear error messages without any
 * out-of-bounds access.
 */

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class p4_fc {

    private static final String[] VEHICLES = {
            "Car",
            "Truck",
            "Bicycle",
            "Motorcycle",
            "Bus"
    };

    public static void main(String[] args) {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int numVehicles = VEHICLES.length;

        System.out.print("Enter the index of a vehicle (0-" + (numVehicles - 1) + "): ");
        System.out.flush();

        String line;
        try {
            line = reader.readLine();
        } catch (IOException e) {
            System.err.println("Error: failed to read input.");
            return;
        }

        if (line == null) {
            System.err.println("Error: no input provided.");
            return;
        }

        line = line.trim();

        int index;
        try {
            index = Integer.parseInt(line);
        } catch (NumberFormatException e) {
            System.err.println("Error: input is not a valid integer.");
            return;
        }

        if (index < 0 || index >= numVehicles) {
            System.err.println("Error: index " + index + " is out of range (0-"
                    + (numVehicles - 1) + ").");
            return;
        }

        System.out.println("Selected vehicle: " + VEHICLES[index]);
    }
}

File: javascript/gpt/p4_fc.js
/**
 * p4_fc.js - Safe vehicle index lookup (JavaScript, Node.js)
 *
 * Prompts the user to enter the index of a vehicle, reads input safely as a
 * string, attempts integer conversion, validates bounds, and prints the
 * corresponding vehicle from a predefined array. Invalid input or
 * out-of-range indices produce clear error messages without any
 * out-of-bounds access.
 */

const VEHICLES = ["Car", "Truck", "Bicycle", "Motorcycle", "Bus"];

function main() {
  const numVehicles = VEHICLES.length;
  process.stdout.write(`Enter the index of a vehicle (0-${numVehicles - 1}): `);

  const fs = require("fs");
  let input;
  try {
    input = fs.readFileSync(0, "utf8");
  } catch (err) {
    console.error("Error: failed to read input.");
    process.exit(1);
    return;
  }

  if (!input) {
    console.error("Error: no input provided.");
    process.exit(1);
    return;
  }

  const line = input.trim();

  if (!/^[+-]?\d+$/.test(line)) {
    console.error("Error: input is not a valid integer.");
    process.exit(1);
    return;
  }

  const index = Number.parseInt(line, 10);

  if (!Number.isInteger(index)) {
    console.error("Error: parsed value is not an integer.");
    process.exit(1);
    return;
  }

  if (index < 0 || index >= numVehicles) {
    console.error(
      `Error: index ${index} is out of range (0-${numVehicles - 1}).`
    );
    process.exit(1);
    return;
  }

  console.log(`Selected vehicle: ${VEHICLES[index]}`);
}

if (require.main === module) {
  main();
}

